### ISSUE TITLE
Multiple reports of usernames and passwords showing up in URLS

### ISSUE THREAD

**Issue by DarrenMcRoy** [zooniverse/Panoptes#1189]
Over the weekend, I got three reports of people saying that when they logged in, they were seeing their login name and password appear in the URL bar, and were concerned about that.

I've asked all three for browser, device, and OS info.

**From user SBolt89:**

Hi! The new homepage looks quite nice! I have one little thing however, and that is the fact that *my personal login and password are part of the URL in my browser.* Is this a result of the new website implementation, and, if so, can displaying my password as part of the URL be mitigated? Thanks in advance!

**From user with email davies.cameronj@gmail.com:**

I noticed when I logged in today that *the username and password were shown in the address bar* as zooniverse.org/login=xxxx&password=xxxx haha not the best thing to see pop up! I’m home and on personal computer, but thought you should know.

**From user with email justinpags@gmail.com:**

I found a security issue on the Zooniverse site.

Basically, I just went to reset my password using this link: https://www.zooniverse.org/#/reset-password

After resetting it, *the website automatically redirected to this link (I replaced my username, password, and the reset token with asterisks)*: https://www.zooniverse.org/?login=***************&password=***************#/reset-password?reset_password_token=***************

From there my username and password continued to show up the URL when I browsed to other parts of the site, e.g., https://www.zooniverse.org/?login=***************&password=***************#/collections/

**Comment by DarrenMcRoy** [comment]
Heard back from one person (davies.cameronj@gmail.com): Google Chrome on Windows 8.

Doesn't seem to be happening for me on Google Chrome (though I'm on Mac).

**Comment by camallen** [comment]
I can't replaciate this behaviour via password reset, CAS login, zooniverse.org login nor ouroboros login. @DarrenMcRoy can you please get more information from the reporting users on how to replicate this? 

@edpaget @parrish @brian-c thoughts on this one?

**Comment by eatyourgreens** [comment]
@camallen we can try this on Windows 8 when I get into the office.

**Comment by eatyourgreens** [comment]
Just tried Chrome 43 on Win 8 and couldn't replicate this by logging in to zooniverse.org.

**Comment by DarrenMcRoy** [comment]
I can't replicate either, but it's definitely a real thing, and definitely recent, as I only started hearing about it over the weekend.

Someone shared a screenshot:

![mail google com](https://cloud.githubusercontent.com/assets/8987864/8905998/575bc42e-3430-11e5-912d-239d82f69afe.png)

@camallen What info would be most helpful in diagnosing this?

**Comment by camallen** [comment]
Any steps the user took / takes to recreate the issue. Did they follow links? Button presses and form submissions to create the URL you see in the screenshot above.

**Comment by astopy** [comment]
@eatyourgreens suggested that this could be due to there being no default method/action on the login form. That sounds plausible to me; if the user gets a JS error or for some reason their browser fails to prevent the default action then it will submit the login form with a GET request. I've reproduced this by filling in the login form, then disabling javascript, then clicking the sign in button.

**Comment by camallen** [comment]
@astopy neat work, i can replicate the URL behaviour with JS disabled. 

Not sure how they get the page to load / login form to appear if JS is disabled but this may be on the right track.

**Comment by edpaget** [comment]
Yeah so there are a couple ways to fix this (in the Front-End, it's not a panoptes issue). 

+ Change the button to a `type="button"` and use an onClick handler instead of the form's onSubmit handler
+ Change the form method to "post" so if something JS-related fails it won't put the username/password in the location bar.

I'd like to emphasize this is only a security risk for the user if someone is looking over their shoulder. The query params are still encrypted when they're sent to our servers.

**Comment by DarrenMcRoy** [comment]
I heard back from justinpags, as seen above:

"I was using Chrome on Windows 10. It was the first time I had visited the site in a long time, so I probably didn't have any cookies set for it. Maybe that condition results in credentials being stored in URL parameters when someone does a password reset?"

It looks like you guys have your finger on the pulse. It's probably not a really serious security issue, as @edpaget just pointed out, but I can definitely see it being alarming to users. :-)

**Comment by camallen** [comment]
Bookmarks could be a problem too.

**Comment by astopy** [comment]
@edpaget It could also be a security risk if their browser leaks the URL in the referrer header...

**Comment by eatyourgreens** [comment]
Setting a `method` (and maybe `action` too) on the form tag at
https://github.com/zooniverse/Panoptes-Front-End/blob/master/app/partials/sign-in-form.cjsx#L28
would probably fix this. It's a good idea to set sensible defaults for your forms in HTML anyway.

It could be caused by `e.preventDefault()` not running, in the `handleSubmit` method. Default behaviour for forms is method=GET, which would reproduce this bug.

Perhaps a Chrome extension is causing this? I notice they've got uBlock running, and a couple of other extensions.

Wouldn't it be a serious security issue since eg. Google Analytics can see the URL params and pass them back to Google? Plus any extensions that are running. Imagine if you inadvertently tweeted a link while your username/password were in the URL.

**Comment by edpaget** [comment]
Closed by https://github.com/zooniverse/Panoptes-Front-End/pull/1231

**Comment by camallen** [comment]
@DarrenMcRoy when this is deployed can you please get your user to see if they can replicate this behaviour?

**Comment by DarrenMcRoy** [comment]
You got it @camallen -- send me a Slack message wen you're sure it's deployed?
